PackageExplorer/Controls/MultiSelectTreeView.cs

```diff
@@ -0,0 +1,291 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Media3D;
+
+namespace PackageExplorer.Controls
+{
+    public class MultiSelectTreeView : TreeView
+    {
+        public MultiSelectTreeView()
+        {
+            GotFocus += OnTreeViewItemGotFocus;
```

Comment:
Maybe use this pattern to properly unsubscribe from the events?
https://stackoverflow.com/questions/25005355/pattern-to-unsubscribe-events-in-customcontrols

Reply:
Used pattern which you recommended :)